### PR TITLE
Fix D&D image inside table

### LINKED ISSUE #4952: Moving image to another cell modifies the target cell
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Open `source` and paste it: 
```
 <table border="1" cellpadding="1" cellspacing="1" style="width:500px">
	<tbody>
		<tr>
			<td><img alt="" src="https://via.placeholder.com/100" /></td>
			<td>&nbsp;</td>
		</tr>
		<tr>
			<td>&nbsp;</td>
			<td>&nbsp;</td>
		</tr>
	</tbody>
</table>
```
2. Using drag & drop move the image into another cell.

### Expected result

The image was moved properly.

### Actual result

Instead of moving the image itself, a new nested table with the image was created in the target.
This problem occurred since `4.7.x` version.

This problem won't occur when we add an empty space after the image and without `clipboard` plugin.

https://user-images.githubusercontent.com/18471998/140518462-c9c8f6c2-e5b2-44ce-a2cf-7a432ebc2273.mov

## Other details

* Browser: …
* OS: …
* CKEditor version: since 4.7.x
* Installed CKEditor plugins: …

### PR DESCRIPTION
**Pull request by KarolDawidziuk**
## What is the purpose of this pull request?

Bug fix.

## Does your PR contain necessary tests?

All patches that change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Manual tests

## Did you follow the CKEditor 4 code style guide?

Your code should follow the guidelines from the [CKEditor 4 code style guide](https://github.com/ckeditor/ckeditor4/blob/major/dev/docs/codestyle.md) which helps keep the entire codebase consistent.

- [x] PR is consistent with the code style guide

## What is the proposed changelog entry for this pull request?

```
* [#4952](https://github.com/ckeditor/ckeditor4/issues/4952): Fix: Moving image to another cell modifies the target cell.
```

## What changes did you make?

There was a problem with [`isSelectedCell()`](https://github.com/ckeditor/ckeditor4/blob/master/core/editor.js#L690) inside `createDocumentFragmentFromRanges()` function, which checks for an invalid value in case we try to DnD an image that is the only element in the cell. 
The fix was relatively simple, consisting of adding an additional condition to check if the range we want to move is in `<td>` and contains only one `` element.

Also, I moved the body from the if condition into separate variables for easier understanding. 

## Which issues does your PR resolve?

Closes #4952.

### CODE REVIEW

**Comment by sculpt0r** [comment]
Extracted from: https://github.com/ckeditor/ckeditor4/pull/4951

File: core/editor.js [hunk 1]
```diff
@@ -684,10 +684,13 @@
 		// 2. <td>[Cell]</td> (IE8-, Safari)
 		function isSelectedCell( range ) {
 			var start = range.startContainer,
-				end = range.endContainer;
+				end = range.endContainer,
+				startIsTr = start.is && start.is( 'tr' ),
+				startIsTdWithEqualChildCount = start.is && start.is( 'td' ) && start.equals( end ) && range.endOffset === start.getChildCount(),
```
Comment (sculpt0r) [inline core/editor.js:689]: As you made `starIsTr` - the `startIsTd` seems to be also useful. So we will get rid of repetition and receive shorter lines of code.

File: tests/plugins/clipboard/manual/draganddropimageintable.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+@bender-tags: 4.17.2, bug, 4952
```
Comment (sculpt0r) [inline tests/plugins/clipboard/manual/draganddropimageintable.md:1]: Why don't we o put those tests also in `tableselection` plugin? The source issue have tests in placed there 🤔

File: tests/plugins/clipboard/manual/draganddropimageintable.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<textarea id="editor" >
+	<table border="1" cellpadding="1" cellspacing="1" style="width:500px">
+		<tbody>
+			<tr>
+				<td><img alt="" src="https://via.placeholder.com/100" /></td>
+				<td>&nbsp;</td>
```
Comment (sculpt0r) [inline tests/plugins/clipboard/manual/draganddropimageintable.html:6]: Those `&nbsp;` makes me suspicious, especially that the related issue is about the extra space after the image ;)

Seems working properly without them, so please remove them from the content.

**Review by sculpt0r — CHANGES_REQUESTED** [review]
Almost there within the first review - nice 👏   ;)

Please take a look at the inline comments.

File: tests/plugins/clipboard/manual/draganddropimageintable.md (hunk 2 above)
Comment (KarolDawidziuk) [reply tests/plugins/clipboard/manual/draganddropimageintable.md:1]: if we disable the`clipboard` plugin, the problem with DnD images in the table disappears, that's why I put this in the clipboard manual tests. I'll move it :)

**Comment by KarolDawidziuk** [comment]
@sculpt0r Thank you for the review.

I've made the improvements you mentioned and it's ready for another one.

**Review by sculpt0r — APPROVED** [review]
Thank you @KarolDawidziuk for corrects 👍 

I slightly adjust the proposed changelog entry.

**Pull request merged**